src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final int DEFAULT_MIN_INCREMENT = 1;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        validateAddIndex(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        ensureCapacity();
+        System.arraycopy(elementData, index, elementData, index + 1, size);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        ensureCapacity(list.size());
+        for (int i = 0; i < list.size(); i++) {
+            /*add() is not used to avoid calling ensureCapacity() in every iteration*/
```

Comment:
most of the times the execution of ensureCapacity will end with size check - it's better to reuse it than maintaining the redundant code.